lib/opFns.js

```diff
@@ -122,22 +122,22 @@ module.exports = {
   },
   // 0x10 range - bit ops
   LT: function (a, b, runState) {
-    return new BN(a.cmp(b) === -1)
+    return new BN(a.cmp(b) === -1 ? 1 : 0)
   },
   GT: function (a, b, runState) {
-    return new BN(a.cmp(b) === 1)
+    return new BN(a.cmp(b) === 1 ? 1 : 0)
   },
   SLT: function (a, b, runState) {
-    return new BN(a.fromTwos(256).cmp(b.fromTwos(256)) === -1)
+    return new BN(a.fromTwos(256).cmp(b.fromTwos(256)) === -1 ? 1 : 0)
   },
   SGT: function (a, b, runState) {
-    return new BN(a.fromTwos(256).cmp(b.fromTwos(256)) === 1)
+    return new BN(a.fromTwos(256).cmp(b.fromTwos(256)) === 1 ? 1 : 0)
   },
   EQ: function (a, b, runState) {
-    return new BN(a.cmp(b) === 0)
+    return new BN(a.cmp(b) === 0 ? 1 : 0)
   },
   ISZERO: function (a, runState) {
-    return new BN(a.isZero())
+    return new BN(a.isZero() ? 1 : 0)
```

Comment:
I think these make sense, but were there actually causing an issue?

Reply:
Yes, creating a ``BN`` instance from ``true`` is actually resulting in another value initialization than with ``1``. 

```
> new BN(true)
<BN: 7d0e>
> new BN(1)
<BN: 1>
```

Not sure if this is an error in ``bn.js``. I find this more explicit and better to read anyway.

Reply:
True, we are till using an outdated bn.js version which treated everything as a number and did random things. Finally a recent version would throw an error of invalid digit for given number base :)